web/styles/search.css

```diff
@@ -2,150 +2,232 @@
+                    0 1px 5px var(--color-search-box-hover-shadow);
+            }
+
+            .search_icon {
+                cursor: pointer;
+            }
+        }
+
         @media (width < $md_min) {
-            margin-left: 40px;
+            #searchbox-input-container {
+                width: 40px;
+            }
+
+            .search-input {
+                opacity: 0;
+            }
+        }
+
+        @media (width < $sm_min) {
+            #searchbox-input-container .search_icon {
+                font-size: 18px;
+            }
         }
     }
 
-    .input-append {
-        position: relative;
+    .navbar-search.expanded {
+        color: hsl(0deg 0% 0% / 100%);
         width: 100%;
-        width: max-content;
-        min-width: 100%;
-
-        .zulip-icon-search {
-            padding: 0;
-            font-size: 20px;
-            position: absolute;
-            left: 10px;
-            top: 10px;
-            z-index: 5;
-
-            @media (width < $sm_min) {
-                top: 5px;
+        box-shadow:
+            0 4px 20px hsl(0deg 0% 0% / 25%),
+            0 1px 5px hsl(0deg 0% 0% / 10%);
+
+        @media (width < $md_min) {
+            /* 3px chosen so that the cursor clicks the search button
+               and close button from the same position. */
+            width: calc(100% - 3px);
+            /* z-index to not see the gear icon underneath */
+            z-index: 1;
+        }
+
+        .search-input {
+            color: var(--color-text-search);
+
+            &::placeholder {
+                color: hsl(0deg 0% 0%);
             }
         }
+    }
 
-        .zulip-icon-search:not(.deactivated) {
-            cursor: pointer;
+    .typeahead.dropdown-menu {
+        /* Match the typeahead's width to its parent container. */
+        right: 0;
+        top: 0;
+        margin-top: 0;
+        padding-top: 0;
+        box-shadow: none;
+        position: relative;
+        width: 100%;
+        background: var(--color-background-search);
+        color: var(--color-text-search);
+
+        .active > a {
+            background-color: var(--color-background-search-option-hover);
+            background-image: none;
+            color: var(--color-text-search-hover);
         }
+        border-width: 0;
     }
 
-    #search_query {
+    .input-append {
+        position: relative;
         width: 100%;
-        font-size: 16px;
-        height: calc(var(--header-height) - 1px);
-        min-height: 30px;
+        margin-bottom: 0;
+    }
+
+    .search-input {
+        font-size: 14px;
+        /* override bootstrap padding for input[type="text"] */
         padding: 0;
-        padding-right: 40px;
         border: none;
+        outline: none;
         border-radius: 0;
         font-family: "Source Sans 3 VF", sans-serif;
-        font-weight: 300;
-        line-height: var(--header-height);
+        font-weight: 400;
+        height: 28px;
```

Comment:
This 28px height shows up a couple of places nearby. Might also be good to put in a variable, like `--header-height` above, which seems to be gone now?

Reply:
header height is still around!

<img width="355" alt="image" src="https://github.com/zulip/zulip/assets/5634097/ff53295d-041a-4aed-9f89-77bccafc058e">

Added this

```css
    /*
    Height of the search box, which appears in the header.
    On very small screens, the search box's height becomes
    the height of its parent.
    */
    --search-box-height: 28px;
```